warehouse/models/mart/gtfs/fct_monthly_shapes.sql

```diff
@@ -0,0 +1,70 @@
+{{ config(materialized='table') }}
+
+WITH dim_gtfs_datasets AS (
+    SELECT *
+    FROM {{ ref('dim_gtfs_datasets') }}
+),
+
+fct_scheduled_trips AS (
+    SELECT *
+    FROM {{ ref('fct_scheduled_trips') }}
+    WHERE service_date >= '2023-07-29' AND service_date <= '2023-07-31'
+),
+
+dim_shapes_arrays AS (
+
+    SELECT
+        key,
+        base64_url,
+        shape_id,
+        pt_array
+
+    FROM {{ ref('dim_shapes_arrays') }}
+    WHERE {{ incremental_where(default_start_var='GTFS_SCHEDULE_START',
+                               this_dt_column='_feed_valid_from',
+                               filter_dt_column='_feed_valid_from',
+                               dev_lookback_days = None)
+           }}
+),
+
+trips_by_shape AS (
+
+    SELECT
+        base64_url,
+        route_id,
+        shape_id,
+        EXTRACT(month FROM service_date) AS month,
+        EXTRACT(year FROM service_date) AS year,
+        COUNT(*) AS n_trips,
```

Comment:
I would use something like this: https://github.com/cal-itp/data-infra/blob/main/warehouse/models/mart/gtfs_quality/idx_monthly_reports_site.sql#L15 to get the last date of the given month as well (`LAST_DAY(service_date, MONTH) AS month_last_day`), so you can use that individual date for the join with the shapes and use that to filter out incomplete months